include/aws/iotdevice/private/secure_tunneling_impl.h

```diff
@@ -223,6 +243,7 @@ struct aws_secure_tunnel {
 
     /* Stores what has been received but not processed */
     struct aws_byte_buf received_data;
+    struct aws_mutex received_data_lock;
```

Comment:
I definitely set up the requested_event_loop in the aws_websocket_client_connection_options. I think what happened was I was running into a number of race conditions in the tests where I'm using a mock websocket instead of a real one. I'll look seeing if I can't get that test to run on the same event loop and remove this mutex locking.